include/vara/Feature/FeatureModelTransaction.h

```diff
@@ -232,35 +555,101 @@ class FeatureModelCopyTransactionBase {
     if (Parent) {
       // To correctly add a parent, we need to translate it to a Feature in
       // our copied FeatureModel
-      Feature *TranslatedParent = FM->getFeature(Parent->getName());
       return FeatureModelModification::make_modification<AddFeatureToModel>(
-          std::move(NewFeature), TranslatedParent)(*FM);
+          std::move(NewFeature), TranslateFeature(*Parent))(*FM);
     }
 
     return FeatureModelModification::make_modification<AddFeatureToModel>(
         std::move(NewFeature))(*FM);
   }
 
+  Relationship *addRelationshipImpl(Relationship::RelationshipKind Kind,
+                                    FeatureVariant Parent) {
+    if (!FM) {
+      return nullptr;
+    }
+
+    return FeatureModelModification::make_modification<AddRelationshipToModel>(
+        Kind, TranslateFeature(*AddRelationshipToModel::resolveVariant(
+                  *FM, std::move(Parent))))(*FM);
+  }
+
+  void addLocationImpl(const FeatureVariant &F, FeatureSourceRange FSR) {
+    assert(FM && "");
+
+    FeatureModelModification::make_modification<AddLocationToFeature>(
+        F, std::move(FSR))(*FM);
+  }
+
+  Constraint *addConstraintImpl(std::unique_ptr<Constraint> NewConstraint) {
+    if (!FM) {
+      return nullptr;
+    }
+
+    return FeatureModelModification::make_modification<AddConstraintToModel>(
+        std::move(NewConstraint))(*FM);
+  }
+
+  void setNameImpl(std::string Name) {
+    assert(FM && "");
+
+    FeatureModelModification::make_modification<SetName>(std::move(Name))(*FM);
+  }
+
+  void setCommitImpl(std::string Commit) {
+    assert(FM && "");
+
+    FeatureModelModification::make_modification<SetCommit>(std::move(Commit))(
+        *FM);
+  }
+
+  void setPathImpl(fs::path Path) {
+    assert(FM && "");
+
+    FeatureModelModification::make_modification<SetPath>(std::move(Path))(*FM);
+  }
+
+  RootFeature *setRootImpl(std::unique_ptr<RootFeature> Root) {
+    if (!FM) {
+      return nullptr;
+    }
+
+    return FeatureModelModification::make_modification<SetRoot>(
+        std::move(Root))(*FM);
+  }
+
+  static void addChildImpl(const FeatureTreeNodeVariant &Parent,
+                           const FeatureTreeNodeVariant &Child) {
+    FeatureModelModification::make_modification<AddChild>(Parent, Child);
+  }
+
 private:
   std::unique_ptr<FeatureModel> FM;
+
+  [[nodiscard]] Feature *TranslateFeature(Feature &F) {
+    return FM->getFeature(F.getName());
+  }
```

Comment:
```suggestion
  [[nodiscard]] Feature *TranslateFeature(Feature &F) {
    return FM->getFeature(F.getName());
  }
  
  std::unique_ptr<FeatureModel> FM;
```
Member vars should be last